lib/puppet/type/file.rb

```diff
@@ -881,6 +885,10 @@ def write(property = nil)
 
         fail_if_checksum_is_wrong(file.path, content_checksum) if validate_checksum?
         if self[:validate_cmd]
+          file.chmod(mode_int) unless mode_int.nil?
```

Comment:
we still need to account for windows here. Do we even attempt to manage mode or ownership when on windows?

Reply:
@Iristyle @joshcooper thoughts on ^^ ?

Reply:
I would call `Puppet::FileSystem.chmod(mode_int, file.path) if mode_int.nil?` here. This way if a user decides to manage a file resource's mode, then the tempfile will have the same permissions.